FRENCH/src/ch17-02-trait-objects.md

```diff
@@ -0,0 +1,1021 @@
+when our code is compiled.
+-->
+
+Au chapitre 8, nous avions mentionné qu'une limite des vecteurs est qu'ils ne
+peuvent stocker des éléments que d'un seul type. Nous avions contourné le
+problème dans l'encart 8-10 en définissant une énumération `Cellule` avec des
+variantes pouvant contenir des entiers, des flottants et du texte. Ainsi, on
+pouvait stocker différents types de données dans chaque cellule et quand même
+avoir un vecteur qui représentait une rangée de cellules. C'est une très bonne
+solution quand nos éléments interchangeables ne possèdent qu'un ensemble bien
+déterminé de types que nous connaissons lors de la compilation de notre code.
+
+<!--
+However, sometimes we want our library user to be able to extend the set of
+types that are valid in a particular situation. To show how we might achieve
+this, we’ll create an example graphical user interface (GUI) tool that iterates
+through a list of items, calling a `draw` method on each one to draw it to the
+screen—a common technique for GUI tools. We’ll create a library crate called
+`gui` that contains the structure of a GUI library. This crate might include
+some types for people to use, such as `Button` or `TextField`. In addition,
+`gui` users will want to create their own types that can be drawn: for
+instance, one programmer might add an `Image` and another might add a
+`SelectBox`.
+-->
+
+Cependant, nous avons parfois envie que l'utilisateur de notre bibliothèque
+puisse étendre l'ensemble des types valides dans une situation donnée. Pour
+montrer comment nous pourrions y parvenir, créons un exemple d'outil d'interface
+graphique (GUI) qui itère sur une liste d'éléments et appelle une méthode
+`afficher` sur chacun d'entre eux pour l'afficher à l'écran — une technique
+courante pour les outils d'interface graphique. Créons une *crate* de
+bibliothèque appelée `gui` qui contient la structure d'une bibliothèque
+d'interface graphique. Cette *crate* pourrait inclure des types que les usagers
+pourront utiliser, tels que `Bouton` ou `ChampDeTexte`. De plus, les
+utilisateurs de `gui` voudront créer leurs propres types qui pourront être
+affichés : par exemple, un développeur pourrait ajouter une `Image` et un autre
+pourrait ajouter une `ListeDeroulante`.
+
+<!--
+We won’t implement a fully fledged GUI library for this example but will show
+how the pieces would fit together. At the time of writing the library, we can’t
+know and define all the types other programmers might want to create. But we do
+know that `gui` needs to keep track of many values of different types, and it
+needs to call a `draw` method on each of these differently typed values. It
+doesn’t need to know exactly what will happen when we call the `draw` method,
+just that the value will have that method available for us to call.
+-->
+
+Nous n'implémenterons pas une véritable bibliothèque d'interface graphique pour
+cet exemple, mais nous verrons comment les morceaux pourraient s'assembler. Au
```

Comment:
Rather `comment les différents éléments pourraient s'assembler`.
I think using *conditionnel* is necessary here because the original text uses `would` and it refers to an alternate reality where we would actually implement a GUI library.